src/Harmony.php

```diff
@@ -126,11 +138,10 @@ protected function executeCondition(array $conditionArray): void
             return;
         }
 
-        $harmony = new Harmony($this->request, $this->response);
-        $callable($harmony, $this->request);
-        $harmony->run();
+        $this->injectNewMiddlewarePosition = $this->currentMiddleware + 1;
+        $callable($this, $this->request);
+        $this->injectNewMiddlewarePosition = null;
 
-        $this->request = $harmony->request;
-        $this->response = $harmony->response;
+        $this->handle($this->request);
```

Comment:
Wouldn't just adding this line (and removing the other changes) fix the issue?

Reply:
No, I've tried.

Reply:
I also tried it, and it worked for me, even with nested conditions.

Reply:
The nice thing about the whole PSR is the ability to stop all further middleware from processing. This is not the case with this workaround. It will always continue outside the conditions.

You could consider building an array with inject positions in my fix. In my new implementation I allow pushing and unshifting. All processed middleware is shifted from the middleware queue, so from within a condition unshifting, places the new middleware at the top op the queue. Also the request and response properties are obsolete in this way, and the counter to keep the position.

But hey, there are multiple ways to skin a cat:)

Reply:
The nice thing about the whole PSR is the ability to stop all further middleware from processing. This is not the case with this workaround. It will always continue outside the conditions.

You could consider building an array with inject positions in my fix. In my new implementation I allow pushing and unshifting. All processed middleware is shifted from the middleware queue, so from within a condition unshifting, places the new middleware at the top op the queue. Also the request and response properties are obsolete in this way, and the counter to keep the position.

But hey, there are multiple ways to skin a cat:)